onmt/decoders/decoder.py

```diff
@@ -612,13 +384,19 @@ class DecoderState(object):
 
     Modules need to implement this to utilize beam search decoding.
     """
+    # def detach(self):
+    #    """ Need to document this VN """
+    #    for h in self._all:
+    #        if h is not None:
+    #            h.detach_()
 
     def detach(self):
-        for h in self._all:
-            if h is not None:
-                h.detach_()
+        """ Need to document this """
```

Comment:
Is this a change in behavior? Also is this a note to us, Maybe add a "?" or a "TODO:" ?

Reply:
@pltrdy  I don't recall exactly. This was to fix the transformer behavior while migrating to 0.4
But not only.